admin/services/attendance.service.js

```diff
@@ -64,6 +64,24 @@ const attendanceService = {
    */
   getAttendanceCodes: async () => {
     return attendanceCodeDataService.sqlFindAttendanceCodes()
+  },
+
+  /**
+   * Check if pupil has an existing attendance entry for the
+   * specific `pinEnv` environment.
+   *
+   * All attendance entries are considered for live checks, while for
+   * familiarisation checks, only `left school` entries are considered
+   *
+   * @param pupilId
+   * @param pinEnv
+   * @return {boolean}
+   */
+  hasAttendance: async (pupilId, pinEnv) => {
+    const pupilAttendance = await pupilAttendanceDataService.findOneByPupilId(pupilId)
+    return pinEnv === 'live'
+      ? pupilAttendance && pupilAttendance.attendanceCode_id !== undefined
+      : pupilAttendance && pupilAttendance.attendanceCode_id === 2 // left school
```

Comment:
Is it intentional to only take into account `left school` attendance id (2) as attendance for non live environments?

Reply:
Yes... but, thinking about it, it is a magic number, which is not great. There's no constant in the codebase, but I've updated the query that checks the user code to also do a join and get the code, and changed the codebase to check the code instead of the ID.